packages/core/src/rum/instrumentation/resourceTracking/implementation/XHRProxy.ts

```diff
@@ -0,0 +1,187 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+import Timer from '../../../../Timer';
+import { DdRum } from '../../../../foundation';
+import type { DdRumXhr } from '../../DdRumXhr';
+import type { RequestProxyOptions } from '../domain/interfaces/RequestProxy';
+import { RequestProxy } from '../domain/interfaces/RequestProxy';
+
+import { URLHostParser } from './URLHostParser';
+import { getTracingAttributes } from './distributedTracing';
+import { createTimings } from './resourceTiming';
+import { calculateResponseSize } from './responseSize';
```

Comment:
Ideally, we would never import from same-level implementation here.

Either it is a sub part of the implementation (calculateResponseSize, calculateResponseSize) for things that are clearly dependent from this implementation and only internal - i.e. no interface in Domain.
Or it is passed as a provider in the constructor (getTracingAttributes, createTimings) for things that are related to something else and that deserve an interface in the domain.

However, I believe using the provider makes us lose the "click to see definition" which is great for one-time contributors.

I think we can achieve the same by having interfaces for top-level files and have a file structure for internal implementation.